# **problem 1**

In [ ]:
employees = {
    "Ravi": 92,
    "Anita": 88,
    "Kiran": 92,
    "Suresh": 85
}
high_score = max(employees.values())
top_employees = []
for name, score in employees.items():
    if score == high_score:
        top_employees.append(name)
print("Top Performers Eligible for Bonus:", ", ".join(top_employees), f"(Score: {high_score})")

Top Performers Eligible for Bonus: Ravi, Kiran (Score: 92)


# **problem 2**

In [ ]:
S = "Buy mobile phone buy phone online"
query = S.lower()

#split into words
words = query.split()

#count words
word_count = {}
for word in words:
    if word in word_count:
        word_count[word] += 1
    else:
        word_count[word] = 1

# keep only words with count > 1
result = {}
for word, count in word_count.items():
    if count > 1:
        result[word] = count

print(result)

{'buy': 2, 'phone': 2}


# **problem  3**

In [ ]:
readings = [3, 4, 7, 8, 10, 12, 5]

valid_readings = []

for hour, value in enumerate(readings):
    if value % 2 == 0:   # check if even
        valid_readings.append((hour, value))

print("Valid Sensor Readings (Hour, Value):")
print(valid_readings)

Valid Sensor Readings (Hour, Value):
[(1, 4), (3, 8), (4, 10), (5, 12)]


# **problem 4**

In [1]:
emails = [
    "ravi@gmail.com",
    "anita@yahoo.com",
    "kiran@gmail.com",
    "suresh@gmail.com",
    "meena@yahoo.com"
]
domains = [email.split("@")[1] for email in emails]
domain_counts = {}
for d in domains:
    domain_counts[d] = domain_counts.get(d, 0) + 1
total = len(emails)
for domain, count in domain_counts.items():
    percentage = (count / total) * 100
    print(f"{domain}: {percentage:.0f}%")

gmail.com: 60%
yahoo.com: 40%


# **problem 5**

In [7]:
sales = [1200, 1500, 900, 2200, 1400, 3000]

average = sum(sales) / len(sales)
threshold = 1.3 * average

print("Average sales:", average)
print("Threshold (30% above average):", threshold)

for day, value in enumerate(sales, start=1):
    if value >= round(threshold, -2):
        print(f"Day {day}: {value}")

Average sales: 1700.0
Threshold (30% above average): 2210.0
Day 4: 2200
Day 6: 3000


explaination why we cant get day4 output at first:- Average sales = 10200/6=1700
- Threshold (30% above average) = 1.3\times 1700=2210
Now check Day 4:
- Day 4 sales = 2200
- Threshold = 2210
- Since 2200 is less than 2210, it doesn’t qualify.
That’s why Python doesn’t print Day 4.


# **problem 6**

In [8]:
user_ids = ["user1", "user2", "user1", "user3", "user1", "user3"]
id_cnt = {}

for uid in user_ids:
    id_cnt[uid] = id_cnt.get(uid, 0) + 1

#Print only duplicates (count > 1)
for uid, count in id_cnt.items():
    if count > 1:
        print(f"{uid} → {count} times")

user1 → 3 times
user3 → 2 times
